crates/node/engine/src/task_queue/tasks/build/task.rs

```diff
@@ -322,16 +234,24 @@ impl EngineTaskExt for BuildTask {
             .into());
         }
 
-        // Send the forkchoice update through the input, with the current engine state and the
-        // payload attributes for the block building job.
-        let mut forkchoice = state.create_forkchoice_state();
-        forkchoice.head_block_hash = self.attributes.parent.block_info.hash;
```

Comment:
This is sneaky, but we cannot remove this. The sequencer has the right to reorg the chain at will; We must build all blocks on top of the attributes' parent hash.

By offloading the creation of the `ForkchoiceState` to `ForkchoiceTask`, we don't get this capability.

Reply:
This is also codified in the execution layer: https://github.com/paradigmxyz/reth/blob/HEAD/crates/engine/tree/src/tree/mod.rs#L893-L905